src/auto-evo/simulation/foodsource/HeterotrophicFoodSource.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Linq;
+
+public class HeterotrophicFoodSource : IFoodSource
+{
+    private readonly Compound oxytoxy = SimulationParameters.Instance.GetCompound("oxytoxy");
+
+    private MicrobeSpecies prey;
+    private Patch patch;
+    private float preySpeed;
+    private float preySize;
+    private float totalEnergy;
+
+    public HeterotrophicFoodSource(Patch patch, MicrobeSpecies prey)
+    {
+        this.prey = prey;
+        this.patch = patch;
+        preySpeed = prey.BaseSpeed();
+        preySize = prey.Organelles.Organelles.Sum(organelle => organelle.Definition.HexCount);
+        patch.SpeciesInPatch.TryGetValue(prey, out long population);
+        totalEnergy = population * prey.BaseOsmoregulationCost() * Constants.AUTO_EVO_PREDATION_ENERGY_MULTIPLIER;
+    }
+
+    public float FitnessScore(Species species)
+    {
+        var microbeSpecies = (MicrobeSpecies)species;
+
+        // No canibalism
+        if (species == prey)
+        {
+            return 0.0f;
+        }
+
+        var behaviorScore = microbeSpecies.Aggression / Constants.MAX_SPECIES_AGGRESSION;
+
+        var predatorSize = microbeSpecies.Organelles.Organelles.Sum(organelle => organelle.Definition.HexCount);
+        var predatorSpeed = microbeSpecies.BaseSpeed();
+        predatorSpeed += ProcessSystem
+            .ComputeEnergyBalance(microbeSpecies.Organelles.Organelles.Select(x => x.Definition), patch.Biome,
+                microbeSpecies.MembraneType).FinalBalance;
+
+        // It's great if you can engulf this prey, but only if you can catch it
+        var engulfScore = predatorSize / preySize > Constants.ENGULF_SIZE_RATIO_REQ
+            && !microbeSpecies.MembraneType.CellWall ?
+                Constants.AUTO_EVO_ENGULF_PREDATION_SCORE :
+                0.0f;
+        engulfScore *= predatorSpeed > preySpeed ? 1.0f : 0.1f;
```

Comment:
Why is the value for a faster prey 0.1 and not 0? Is it to simulate that, by chance predators could catch prey?

If yes, 
1. Is it really a good approximation? It's not like a cell would trip. Of course, they could be impeded by chunks or cells (but the predator would as well, and given they are small they would not be really a problem). The most likely case are other predators, but that would be very populations depending then.
2. Shouldn't this apply to the predator as well, as they may miss their prey? (doing like 0.9f : 0.1f).
3. Ultimately, unless there is a formula coming from this, this could use to be defined in Constants.

Reply:
It does simulate the chance of catching prey, in a very inaccurate and un-tested way. There's a few ways to catch something faster than you:
- prey is out of ATP and predator is not
- prey is flanked by two predators
- prey AI makes bad choices 
It also helps a predator not have to run from other things. Not precise, but representative.

The real motive was really to make sure that slightly larger heterotrophs would be encouraged to provide some more challenge in the way of predation. I admit I haven't tested this particular variable in isolation, so maybe it's unnessicary. This is also why I'm reluctant to name a constant here, but I'm not against it.

Reply:
I'd say a constant is always useful to get a better grasp at what things are represented, and they may be tweaked easily from the Constants.cs file. But I won't say that's a necessary change either...